scripts/stats_std.py

```diff
@@ -17,59 +17,58 @@ def mean_of_array(reads,start, pos):
 
-	file = 'output_std.txt'
+			start = 0
+			big_start = 0
+			last_end = 0
 
-	with open(file, "w") as f:
-		print("Index Chrom Pos Thresh Diff", file=f)
-		for i in range(len(reads)):
-			if i == ((len(reads)) - 1):
-				current = reads[i]
-				next_one = reads[0]
-				next_pos = position[0]
-			else:
-				current = reads[i]
-				next_one = reads[i+1]
-				next_pos = position[i+1]
+			for i in range(1000, len(reads), 1000):
+				if i == ((len(reads)) - 1):
+					current = reads[i]
+					next_one = reads[0]
+					next_pos = position[0]
+				else:
+					current = reads[i]
+					next_one = reads[i+1]
+					next_pos = position[i+1]
 
-			if (i % 1000) == 0 and i != 0:
-				#print(str(i) + ' : ' + str(position[i]))
 				mean = mean_of_array(reads, start, i)
 				if mean >= thresh_up:
 					if last_end == start:
-						print(str(big_start) + ' ' + str(i) + ' ' + str(mean) + ' +3sigma ' + str(chrom[big_start]).lstrip('b\'').rstrip('\'') + ':' + str(position[big_start]) + ' ' + str(chrom[i]).lstrip('b\'').rstrip('\'') + ':' + str(position[i]), file=f)
-						print(str(big_start) + ' ' + str(i) + ' ' + str(mean) + ' +3sigma ' + str(chrom[big_start]).lstrip('b\'').rstrip('\'') + ':' + str(position[big_start]) + ' ' + str(chrom[i]).lstrip('b\'').rstrip('\'') + ':' + str(position[i]))
+						big_start_chrom = str(chrom[big_start]).lstrip('b\'').rstrip('\'')
+						i_chrom = str(chrom[i]).lstrip('b\'').rstrip('\'')
+						print(f'{big_start:<15} {i:<15} {mean:<10} +3sigma {big_start_chrom:>20}:{position[big_start]} {i_chrom:>20}:{position[i]}')
+						print(f'{big_start:<15} {i:<15} {mean:<10} +3sigma {big_start_chrom:>20}:{position[big_start]} {i_chrom:>20}:{position[i]}', file=f)
 					else:
-						print(str(start) + ' ' + str(i) + ' ' + str(mean) + ' +3sigma ' + str(chrom[start]).lstrip('b\'').rstrip('\'') + ':' + str(position[start]) + ' ' + str(chrom[i]).lstrip('b\'').rstrip('\'') + ':' + str(position[i]), file=f)
-						print(str(start) + ' ' + str(i) + ' ' + str(mean) + ' +3sigma ' + str(chrom[start]).lstrip('b\'').rstrip('\'') + ':' + str(position[start]) + ' ' + str(chrom[i]).lstrip('b\'').rstrip('\'') + ':' + str(position[i]))
+						start_chrom = str(chrom[start]).lstrip('b\'').rstrip('\'')
+						i_chrom = str(chrom[i]).lstrip('b\'').rstrip('\'')
+						print(f'{start:<15} {i:<15} {mean:<10} +3sigma {start_chrom:>20}:{position[start]} {i_chrom:>20}:{position[i]}')
+						print(f'{start:<15} {i:<15} {mean:<10} +3sigma {start_chrom:>20}:{position[start]} {i_chrom:>20}:{position[i]}', file=f)
 						big_start = start
 					last_end = i
-				else:
-					print(str(start) + ' : ' + str(i) + ' ' + str(position[start]) + ' : ' + str(position[i]) + ' ' + str(mean))
 				start = i
 
-	c1 = subprocess.Popen(["tac {} | sort -u -k 1,1 | tac"].format(file), shell=True, stdout=open("output_std_final.txt", "w"))
+	c1 = subprocess.Popen(["tac {} | sort -u -k 1,1 | tac".format(file)], shell=True, stdout=open("output_std_mgh_contig_final.txt", "w"))
```

Comment:
Is this sorting still necessary?

Reply:
Yes, without the sorting, multiple entries for each region would be there. You can see this if you compare the first text file and the final text file that the code outputs. However, when I merge it with the other code I am going to try and make the first file a temp file.

Reply:
Can you think of a way how to algorithmically prevent the output of multiple entries (maybe using a state variable?)